src/lib/components/chat/ChatInput.svelte

```diff
@@ -19,14 +19,24 @@
 	function handleKeydown(event: KeyboardEvent) {
 		// submit on enter
 		if (event.key === "Enter" && !event.shiftKey && !isCompositionOn) {
-			event.preventDefault();
-			// blur to close keyboard on mobile
-			textareaElement.blur();
-			// refocus so that user on desktop can start typing without needing to reclick on textarea
-			if (isDesktop(window)) {
-				textareaElement.focus();
+			//  using isDesktop to check if the user is on a mobile device
+			if (!isDesktop(window)) {
+				event.preventDefault(); // prevent the default behavior of the enter key
+				// insert a newline at the cursor position
+				const start = textareaElement.selectionStart;
+				const end = textareaElement.selectionEnd;
+				value = value.substring(0, start) + "\n" + value.substring(end);
+				textareaElement.selectionStart = textareaElement.selectionEnd = start + 1;
+			} else {
+				event.preventDefault();
+				// blur to close keyboard on mobile
+				textareaElement.blur();
+				// refocus so that user on desktop can start typing without needing to reclick on textarea
+				if (isDesktop(window)) {
+					textareaElement.focus();
```

Comment:
I think that if statement doesn't need to be there anymore right ? since you're already in the `else` condition here

Reply:
Thanks, you are right I did not catch removing that. Fixed and updated.